virtual_rainforest/models/hydrology/above_ground.py

```diff
@@ -272,3 +272,36 @@ def estimate_interception(
         * (1 - np.exp(-canopy_density_factor * precipitation / max_capacity)),
         nan=0.0,
     )
+
+
+def distribute_monthly_rainfall(
+    total_monthly_rainfall: NDArray[np.float32],
+    num_days: int,
+) -> NDArray[np.float32]:
+    """Distributes total monthly rainfall over the specified number of days.
+
+    At the moment, this function allocates each millimeter of monthly rainfall to a
+    randomly selected day. In the future, this allocation could be based on observed
+    rainfall patterns.
+
+    Args:
+        total_monthly_rainfall: Total monthly rainfall, [mm]
+        num_days: Number of days to distribute the rainfall over
+
+    Returns:
+        An array containing the daily rainfall amounts, [mm]
+    """
+    np.random.seed(42)  # TODO move this to core for all models to be the same
```

Comment:
This feels a bit weird to me in that it means that the function will always have the same result (provided that it is called with the same input), which I don't think is really what you want. Does it make sense to move this seed to the test?

Reply:
It's very likely that I misunderstood, I thought we want simulations to be reproducible and the stochasticity to be somewhat deterministic? I just didn't want to have the exact same rainfall every day. Or do we accept that we always get slightly different results every time we run the vr?

Reply:
Ahh yeah, I guess it's fine for what you are doing at the moment. Just in the end I think we want functions like this to generate a random result, but the overall simulation flow to be reproducible by setting the seed in (I think) `vr_run`

Reply:
Also, this is a legacy function. We probably should be using:

```python
rng = np.random.default_rng(seed=42)
```

And then:

```python
rng.random.randint(0, num_days)
```